mitxgraders/helpers/calc.py

```diff
@@ -542,38 +529,43 @@ def check_variables(self, valid_variables, valid_functions):
         Confirm that all the variables and functions used in the tree are defined.
         Otherwise, raise an UndefinedVariable or UndefinedFunction
         """
-        if self.case_sensitive:
-            casify = lambda x: x
-        else:
-            casify = lambda x: x.lower()  # Lowercase for case insens.
 
-        # Test if casify(X) is valid, but return the actual bad input (i.e. X)
         bad_vars = set(var for var in self.variables_used
-                       if casify(var) not in valid_variables)
+                       if var not in valid_variables)
         if bad_vars:
             message = "Invalid Input: {} not permitted in answer as a variable"
             varnames = ", ".join(sorted(bad_vars))
+
+            # Check to see if there is a different case version of the variable
```

Comment:
Nice. This is just generally helpful, plus it makes me feel better about removing case_sensitive.